xray/default_emitter.go

```diff
@@ -98,17 +100,22 @@ func packSegments(seg *Segment, outSegments [][]byte) [][]byte {
 			cb := ss.StreamCompletedSubsegments(s)
 			outSegments = append(outSegments, cb...)
 		}
-		b, _ := json.Marshal(s)
+		b, err := json.Marshal(s)
+		if err != nil {
+			logger.Errorf("JSON error while marshalling (Sub)Segment: %v", err)
```

Comment:
I think we never check the error when we marshal the segment since the segment is what we create. If we have this error, what should we do? I mean what is the use case for catching and logging this error from the customers' perspective?

Reply:
It is FYI for the customer regarding the marshaling error. Currently due to any reason, if there is marshaling error, say due to unacceptable subsegment name provided by the customer, the entire segment may fail due to marshaling. The customer may wonder why the Segment is not appearing on the X-Ray console. This log will help in debugging the issue.